### DISCUSSION TITLE
Updating Brew Application and App Leftovers

### DISCUSSION THREAD

**Discussion by BelalHmeidat** [Homebrew/discussions#1136] [Casks] [answered]
Hi,
I'm thinking of downloading applications via brew instead of dmg binaries. I have questions about casks installed by brew. Can I update those applications normally by going to the menu bar> application_name> check for updates, or should I always update them by brew upgrade? How are installations via brew different from dmg or pkg? Also, does homebrew remove all leftovers (leftovers in the Library directory for example) when uninstalling an application? 

Thanks,

**Comment by SMillerDev** [comment, marked as answer]
> Can I update those applications normally by going to the menu bar> application_name> check for updates, or should I always update them by brew upgrade?

If they provide an auto updater you can use it
> How are installations via brew different from dmg or pkg?

If it's a cask, they are no different. it's just more convenient to use them with a unified brew interface (with search).

> Also, does homebrew remove all leftovers (leftovers in the Library directory for example) when uninstalling an application?

It tries to, but since we don't write the installers we can't really guarantee anything

**Reply by BelalHmeidat** [reply]
Got it! Thanks a lot!!
